pennylane/ops/op_math/controlled_class.py

```diff
@@ -277,8 +296,9 @@ def map_wires(self, wire_map: dict):
         new_base = self.base.map_wires(wire_map=wire_map)
         new_control_wires = Wires([wire_map.get(wire, wire) for wire in self.control_wires])
         new_work_wires = Wires([wire_map.get(wire, wire) for wire in self.work_wires])
+        cls = self.__class__
 
-        return Controlled(
+        return cls(
```

Comment:
Cosmetic comment: you can call `self.__class__` directly.

Reply:
Updated these